### PR TITLE
docs: add basic docker-compose file and instructions

### PR DESCRIPTION
**Pull request by big-andy-coates**
### Description 

Added a sample docker-compose file with some steps for how to use. Brings up multi-node KSQL with dependants. 

Only outstanding issue is it doesn't look like community members can build docker images locally, as they won't have access to a AWS based docker repo, right?   Does anyone know a way around this, or plans to make this publicly available? cc @apurvam @rodesai @agavra 

### Testing done 

manually checked it all works.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Comment by vinothchandar** [comment]
Public dockerHub? We have stuff here already https://hub.docker.com/r/confluentinc/ksqldb-server

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  schema-registry:
+    image: confluentinc/cp-schema-registry:latest
+    depends_on:
+      - zookeeper
+      - kafka
+    environment:
+      SCHEMA_REGISTRY_HOST_NAME: schema-registry
+      SCHEMA_REGISTRY_KAFKASTORE_CONNECTION_URL: zookeeper:32181
+
+  ksqldb-server-primary:
```
Comment (vinothchandar) [inline docker-compose.yml]: can we just say ksqldb-server-1 and ksqldb-server-2?  We can't really control active/standby assignment and ensure this container will be active always right?

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    hostname: ksqldb-server-primary
+    container_name: ksqldb-server-primary
+    depends_on:
+      - kafka
+      - schema-registry
+    ports:
+      - "8088:8088"
+    environment:
+      KSQL_LISTENERS: http://0.0.0.0:8088
+      KSQL_BOOTSTRAP_SERVERS: kafka:9092
```
Comment (vinothchandar) [inline docker-compose.yml]: Just want to check if this should be 29092? since thats the port that gets exposed for the kafka container?

**Comment by agavra** [comment]
FYI - if you want to build the image without ECR you can change the dockerfile not to depend on `cp-base-new` and just use something open source like the `openjdk:8-jre-alpine` image

**Comment by agavra** [comment]
Also cc @MichaelDrogalis  @derekjn - do we want to have this docker compose in the repo differ from whats on https://ksqldb.io/quickstart.html?

**Comment by derekjn** [comment]
> Also cc @MichaelDrogalis @derekjn - do we want to have this docker compose in the repo differ from whats on https://ksqldb.io/quickstart.html?
I think it's fine if they're different. The quickstart is for getting a minimal development environment up and running. As long as the same images are being used everywhere, I don't think there is a need to maintain uniformity across all of the different `docker-compose` files that may be using these images.

File: docker-compose.yml (hunk 1 above)
Comment (big-andy-coates) [reply docker-compose.yml]: It's not about active / standby. It's about scaling the cluster. 

The primary is at a known address and can't be scaled. The other is at an unknown address and can be scaled.

I originally started with server-1 and server-2, but the command line is better this way around e.g.  `docker-compose up -scale ksqldb-server 3` = 4 node cluster.

Might switch `ksqldb-server` to `additional-ksqldb-server` e.g.  `docker-compose up -scale addtional-ksqldb-server 3`, which is probably more intuitive that its a  4 node cluster.

File: docker-compose.yml (hunk 2 above)
Comment (big-andy-coates) [reply docker-compose.yml]: This is correct.

**Comment by big-andy-coates** [comment]
> Public dockerHub? We have stuff here already https://hub.docker.com/r/confluentinc/ksqldb-server

I'm aware, but its the _build_ that depends on our ECR.

> FYI - if you want to build the image without ECR you can change the dockerfile not to depend on cp-base-new and just use something open source like the openjdk:8-jre-alpine image

Good to know - I'll take a look. However if this something we _want_ to do? I'm not sure of the repercussions...

> Also cc @MichaelDrogalis @derekjn - do we want to have this docker compose in the repo differ from whats on https://ksqldb.io/quickstart.html?

Ah, didn't know about that. We should maybe keep them inline, the difference between the one in this PR and the one in the example is:
 * This one uses `latest` version, docs on uses hardcoded version.  Using `latest` is probably less overhead when versions change.
 * This one defaults to two KSQL nodes and can be scaled: `docker-compose up -scale additional-ksqldb-server 2`.  Not required, but useful.
 * This one sets rf and min isr  on the transaction log, which I believe is required with latest Kafka, so the example on will need updating with this. e.g. 
 * This one depends on SR, which enables Avro format.
 
cc @JimGalasyn

**Comment by JimGalasyn** [comment]
@big-andy-coates This is awesome! @ybyzek may find it useful, too.

**Pull request merged**